.github/workflows/publish.yml

```diff
@@ -0,0 +1,57 @@
+name: Publish
+on:
+  release:
+    types:
+      - published
+
+env:
+  # mark:automatic-version
+  VERSION: "1.20231001.0"
+
+jobs:
+  # TODO
+  # deploy-production:
+
+  push-release-images:
+    # needs: deploy-production
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # TODO
+          # - image_name: client
+          - image_name: relay
+          - image_name: gateway
+    permissions:
+      # Needed to upload artifacts to a release
+      packages: write
+      # Needed to login to GCP
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/gcp-docker-login
+        id: login
+        with:
+          project: firezone-staging
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{github.actor}}
+          password: ${{secrets.GITHUB_TOKEN}}
+      - name: Pull and push
+        env:
+          SRC_IMAGE: ${{ steps.login.outputs.registry }}/firezone/${{ matrix.image_name }}:${{ github.sha }}
+          DST_IMAGES: |
+            ghcr.io/firezone/${{ matrix.image_name }}:${{ github.sha }}
+            ghcr.io/firezone/${{ matrix.image_name }}:${{ env.VERSION }}
+            ghcr.io/firezone/${{ matrix.image_name }}:latest
```

Comment:
@jamilbk please also push them to the production Google Artifact Registry too, here is what I did manually for prod for now:
```
docker tag us-east1-docker.pkg.dev/firezone-staging/firezone/api:1.20231001.0-7034aa3853e1e5d20026e43859a9e9c799d4a110 us-east1-docker.pkg.dev/firezone-prod/firezone/api:1.20231001.0-7034aa3853e1e5d20026e43859a9e9c799d4a110
```

Reply:
Will do. I think this workflow would then be:

1. Pull `relay`, `api`, `web`, `gateway` images from `staging`, push to `prod`
2. Deploy `prod`
3. Push `relay`, `gateway` images to GHCR.io for public to consume

That'll allow the deploy to succeed with prod images on GCP but wait until it's deployed until we publish the `relay` and `gateway`